### ISSUE TITLE
[java] CommentDefaultAccessModifier doesn't recognize /* package */ comment at expected location for constructors

### ISSUE THREAD

**Issue by ax-lothas** [pmd/pmd#5634]
**Affects PMD Version:** 
7.11

**Rule:**
Please provide the rule name and a link to the rule documentation:
CodeStyle.CommentDefaultAccessModifier
https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#commentdefaultaccessmodifier
https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#commentdefaultaccessmodifier

**Description:**
The `/* package */` comment isn't accepted in the expected location.

**Code Sample demonstrating the issue:**

```java
public class Test {

  @SuppressWarnings("")
  /* package */ Test() { // <-- Missing commented default access modifier on constructor 'Test()'
    // nothing
  }

}
```

```java
public class Test {

  /* package */ @SuppressWarnings("")
  Test() { // <-- works, no violation
    // nothing
  }

}
```

```java
public class Test {

  @SuppressWarnings("")
  Test /* package */() { // <-- works, no violation
    // nothing
  }

}
```

**Expected outcome:**

PMD reports a violation at line 4, but that's wrong. That's a false positive.

**Running PMD through:** *Eclipse Plugin, Gradle*
